packages/react-components/src/components/styles/v8StyleShim.ts

```diff
@@ -11,8 +11,10 @@ import { ComponentSlotStyle } from '../../types';
  *
  * @private
  */
-export function createStyleFromV8Style(v8Style: IStyle | ComponentSlotStyle): React.CSSProperties | undefined {
-  const result = {};
+export function createStyleFromV8Style(
+  v8Style: IRawStyle | ComponentSlotStyle | undefined
+): React.CSSProperties | undefined {
+  const result: { [key: string]: React.CSSProperties | undefined } = {};
```

Comment:
I think it should have `React.CSSProperties | undefined` type instead of `{ [key: string]: React.CSSProperties | undefined }`

Reply:
The `[key as string]` is to work around the code that assigns the value:

```
// React.CSSProperties uses camelCase for MS properties but v8Style uses PascalCase
const newRecord = record.substring(0, msSuffix.length).toLowerCase + record.substring(msSuffix.length);
result[newRecord] = v8Style[record];
```
Because the string is used here, the first key type is set to string, rather than the CSSProperties.
Even this:
` result[newRecord as keyof CSSProperties] = v8Style[record];` does not work, as it causes a TypeScipt error:

`Expression produces a union type that is too complex to represent.ts(2590)`

I'd love if if there's a better solution here.